tests/playwright-test/web-server.spec.ts

```diff
@@ -498,6 +498,78 @@ test('should create multiple servers', async ({ runInlineTest }, { workerIndex }
   expect(actualLogMessages).toStrictEqual(expectedLogMessages);
 });
 
+test('should create multiple servers via public API', async ({ runInlineTest }, { workerIndex }) => {
+  const port = workerIndex + 10500;
+  const result = await runInlineTest({
+    'test.spec.ts': `
+        const { test } = pwt;
+
+        test('connect to the server', async ({page}) => {
+          await page.goto('http://localhost:${port}/port');
+          await page.locator('text=${port}');
+
+          await page.goto('http://localhost:${port + 1}/port');
+          await page.locator('text=${port + 1}');
+        });
+      `,
+    'playwright.config.ts': `
+        module.exports = {
+          webServer: [
+            {
+              command: 'node ${JSON.stringify(SIMPLE_SERVER_PATH)} ${port}',
+              url: 'http://localhost:${port}/port',
+            },
+            {
+              command: 'node ${JSON.stringify(SIMPLE_SERVER_PATH)} ${port + 1}',
+              url: 'http://localhost:${port + 1}/port',
+            }
+          ],
+          globalSetup: 'globalSetup.ts',
+          globalTeardown: 'globalTeardown.ts',
+        };
+        `,
+    'globalSetup.ts': `
```

Comment:
Looks like my comment didn't save—startup order is important, and since this new array variant is new/different than what we had before, there's no harm (aside from a bit of repeated code) to ensure it's working under these conditions, too.